tests/integ/test_state_machine_definition.py

```diff
@@ -531,4 +584,56 @@ def test_retry_state_machine_creation(sfn_client, sfn_role_arn, training_job_par
         role=sfn_role_arn
     )
 
-    workflow_test_suite(sfn_client, workflow, asl_state_machine_definition, None)
\ No newline at end of file
+    workflow_test_suite(sfn_client, workflow, asl_state_machine_definition, None)
+
+
+def test_state_machine_creation_with_retry_in_constructor(sfn_client, sfn_role_arn, training_job_parameters):
+    retry_state_name = "RetryStateName"
+    all_fail_error = "Starts.ALL"
+    interval_seconds = 1
+    max_attempts = 2
+    backoff_rate = 2
+    task_resource = f"arn:{get_aws_partition()}:states:::sagemaker:createTrainingJob.sync"
+
+    # change the parameters to cause task state to fail
```

Comment:
specify what will cause it to fail...  i.e. "invalid image will throw badImageexception (or whatever)"